plugins/vpc-bridge/config/k8s/pipe.go

```diff
@@ -0,0 +1,197 @@
+}
+
+// Pipe Represents pipe.
+type Pipe struct {
+	Path       string              // Represents named pipe path.
+	listener   net.Listener        // Represents listener on named pipe.
+	resultChan chan PipeReadResult // Represents results chan to receive output from go routine.
+}
+
+// NewPipe Create Pipe object with random named pipe path.
+func NewPipe() (PipeOperator, error) {
+	// Generate GUID random string.
+	g, err := windows.GenerateGUID()
+	if err != nil {
+		return nil, fmt.Errorf("error creating unique name for named pipe: %w", err)
+	}
+	// Create pipe path with prefix.
+	pipeName := fmt.Sprintf(NamedPipePathPrefix, g.String())
+	return &Pipe{Path: pipeName}, nil
+}
+
+// Start listener on named pipe. Start accepting connections on named pipe in separate go routine.
+// Creates result chan to get output from separate go routine.
+func (p *Pipe) Start(ctx context.Context) error {
+
+	// Dials and start listener on named pipe.
+	listener, err := p.listen()
+	if err != nil {
+		return fmt.Errorf("error creating listener: %w", err)
+	}
+	p.setListener(listener) // Set listener on Pipe object.
+
+	// Creating result channel to get output from pipe.
+	resultChan := make(chan PipeReadResult)
+	p.setResultChan(resultChan) // Set results chan on Pipe object
+
+	// Accept connections and read pipe in go routine as these operations are blocking.
+	go p.acceptConnections(listener, resultChan)
+
+	return nil
+}
+
+// GetPath get name/path for named pipe.
+func (p *Pipe) GetPath() string {
+	return p.Path
+}
+
+// GetOutput Get output received from named pipe using results chan.
+func (p *Pipe) GetOutput(ctx context.Context) (string, error) {
+
+	// Get result chan from Pipe object.
+	resultChan := p.getResultChan()
+
+	// Get output from named pipe using result chan.
+	var result PipeReadResult
+	select {
+	case result = <-resultChan:
+	case <-ctx.Done():
+		return "", fmt.Errorf("timeout getting output from named pipe result chan")
+	}
+	return result.stdout, result.stderr
+}
+
+// Close closes pipe listener.
+func (p *Pipe) Close() error {
+	err := p.getListener().Close()
+	if err != nil {
+		return fmt.Errorf("error closing pipe listener: %w", err)
+	}
+	return nil
+}
+
+// setListener set named pipe listener to Pipe object.
+func (p *Pipe) setListener(listener net.Listener) {
+	p.listener = listener
+}
+
+// getListener get named pipe listener from Pipe object.
+func (p *Pipe) getListener() net.Listener {
+	return p.listener
+}
+
+// setResultChan set results chan to Pipe object.
+func (p *Pipe) setResultChan(outputChan chan PipeReadResult) {
+	p.resultChan = outputChan
+}
+
+// getResultChan get results chan from Pipe object.
+func (p *Pipe) getResultChan() chan PipeReadResult {
+	return p.resultChan
+}
+
+// listen creates named pipe. creates listener on named pipe.
+// Returns listener for named pipe.
+func (p *Pipe) listen() (net.Listener, error) {
+	pipeName := p.GetPath()
```

Comment:
Do this in start to make it more clear you can only do this once

Reply:
done, removed this extra `listen ` func and moved logic to `start()`